### PR TITLE
ci: Add caching to the CI

### PR DESCRIPTION
**Pull request by Milo123459**
This adds caching to the CI, which drastically speeds it up.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -137,6 +140,7 @@ jobs:
         with:
           toolchain: ${{ env.nightly }}
           override: true
+      - uses: Swatinem/rust-cache@v1
```
Comment (Darksonn) [inline .github/workflows/ci.yml:143]: It doesn't sound like this is useful for the tests that use nightly?

Comment (Milo123459) [reply .github/workflows/ci.yml:143]: Why wouldn't it? It can cache nightly and stable builds

Comment (Darksonn) [reply .github/workflows/ci.yml:143]: The nightly cache would usually be out of date, since it changes every night. But I guess that's ok.

**Review by Darksonn — APPROVED** [review]
We'll give this a try.

**Pull request merged**